### DISCUSSION TITLE
Applying clip path to SVG objects

### DISCUSSION THREAD

**Discussion by vkovenko** [pymupdf/PyMuPDF#2872] [Looking for help] [answered]
Greetings. I need to crop pdf along with its SVG. I came up with an approach consisting of the following steps:
1. Get SVG objects using fitz `get_drawings` method.
2. Crop objects by applying geometrical functions.
3. Convert objects back to SVG using svgwrite.
4. Convert SVG to PDF by applying cairosvg.
Everything seemed to work okay, but I found that filling of specific objects with color doesn't work as expected (arrows in black):
Expected result:
<img width="351" alt="expres" src="https://github.com/pymupdf/PyMuPDF/assets/117199553/c470418f-db5a-465b-a749-0379f2ba6c2e">
Actual result:
<img width="315" alt="actres" src="https://github.com/pymupdf/PyMuPDF/assets/117199553/ffcb0ac0-ff8d-4b41-b8d3-158e99e9b9c7">
Then I found that clip paths can be extracted by setting `extended=True ` in `get_drawings` method. I believe that clip paths are used there in order to fill objects with color w.r.t their shape. The question is how to convert those clip paths back to SVG? Maybe you would have some suggestion. Thank you!

**Comment by JorjMcKie** [comment, marked as answer]
Sorry for the tardy answer.
Unfortunately, PyMuPDF does not currently support the output of clipped vector graphics or text.
There also are no plans for the foreseeable future.

**Reply by more-strive** [reply]
May I ask if there is any idea to obtain the clip for image or path，

**Reply by vkovenko** [reply]
Thank you for your response!

**Reply by more-strive** [reply]
May I ask if the author's problem has been resolved

**Reply by vkovenko** [reply]
Yes. I needed to get clip-path from fitz and assign them to specific <line> or <path> inside SVG. I found the other way around for cropping and rotating SVG objects without using fitz for getting parsed SVG objects in python (I still use fitz for getting svg from PDF in a fast manner, but parse objects myself in a custom way using xml, which gives an opportunity to gather all the tags from SVG along with its attributes, including clip-path).

**Reply by more-strive** [reply]
Can you share the demo of this section

**Reply by vkovenko** [reply]
Sorry, but the project I'm working on is under NDA, thus I have no right to share extensive information about the topic.

**Reply by more-strive** [reply]
Okay， [yft-design](https://github.com/dromara/yft-design)  This repository is my open-source image editor. When loading a PDF, there are still images left that cannot be resolved by the clip. If there is any other help, I would be very grateful
